### PR TITLE
werft/VM: Install gitpod on k3s running in Haverster VMs

### PR DESCRIPTION
**Pull request by ArthurSens**
Signed-off-by: ArthurSens <arthursens2005@gmail.com>

## Description (Iteration 1)
This PR points to Mo's branch `me/install_k3s` (See related PR: https://github.com/gitpod-io/gitpod/pull/7444).

It only increases the timeout for retrieving the kubeconfig from the VM. I'm doing that because, as mentioned in https://github.com/gitpod-io/gitpod/pull/7444, SSH is failing. The reason for the failure is just because SSH deamon is taking more than 3 minutes to start, running the job again is enough to retrieve the kubeconfig.

I'm opening this as a draft PR because this is still not enough to have gitpod installed in the k3s cluster. I'm now having problems with certificates created by terraform somewhere on our script.

```release-note
NONE
```

## Description (Iteration 2)
I (Moritz) continued the work on this PR. 

increasing the time out for SSH, unfortunately, did not solve the connection issues, thus I have removed the increase again. See https://github.com/gitpod-io/gitpod/issues/7471 for details. 

I enabled k3s' `local-storage` so that persistent volumes work.

I modified the job to download the docker-pull-secret from the core-dev-cluster before we change the kubectl-context to k3s.

## To test
1. run the job via `werft run github -a with-vm=true`
2.run the job again via `werft run github -a with-vm=true`  (workaround from https://github.com/gitpod-io/gitpod/issues/7471)
3. run `kubectl get pods` on the k3s cluster and see that all pods are in state "running".

### CODE REVIEW

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **geropl** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @geropl` in a comment when ready.

*No associated issue*. Update pull-request body to add a reference to an issue, or get approval with `/approve no-issue`

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/gitpod-io/gitpod/blob/me/install_k3s/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **geropl** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @geropl` in a comment when ready.

*No associated issue*. Update pull-request body to add a reference to an issue, or get approval with `/approve no-issue`

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/gitpod-io/gitpod/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Review by mads-hartmann — APPROVED** [review]
Awesome, this looks good to me 🎉

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: f3258241b6f3748a9c970609438bee1023bf6b1c</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/7445#pullrequestreview-845555144" title="Approved">mads-hartmann</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **geropl** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @geropl` in a comment when ready.

Associated issue: *#7471*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/gitpod-io/gitpod/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by mads-hartmann** [comment]
@meysholdt I added hold label as we need to check that we didn't break normal preview environments before merging :) - Removing the `/werft with-vm=true` from the PR description and just checking that the "old" preview env still works :)

**Comment by meysholdt** [comment]
Triggered the build for the "old" preview envs here:
https://werft.gitpod-dev.com/job/gitpod-build-as-k3s-gitpod.33

Changing the description of the PR did not trigger it :(

**Comment by meysholdt** [comment]
https://werft.gitpod-dev.com/job/gitpod-build-as-k3s-gitpod.35 succeded and I can start workspaces in the "old" preview env. 

merging.

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

Approval requirements bypassed by manually added approval.

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/7445#pullrequestreview-845555144" title="Approved">mads-hartmann</a>*

Associated issue: *#7471*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/gitpod-io/gitpod/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
